### ISSUE TITLE
Implement abstract IBC handshake components in relayer framework

### ISSUE THREAD

**Issue by soareschen** [informalsystems/hermes#2925]
## Summary

Currently in the new relayer, we rely on using the v1 relayer to perform IBC handshakes before relaying the packets. However since relayer v1 only works with Cosmos chains, we cannot use it for establishing IBC connections between a Cosmos and non-Cosmos chain.

We would need re-implement the IBC handshakes in a abstract manner in the relayer framework. Once that is done, we would be able to use it to create new IBC connections between Cosmos chains as well as with non-Cosmos chains.

The implementation of the abstract handshake components can follow the same patterns as the `PacketRelayer`. We will be implementing components such as `ConnectionHandshakeRelayer` and `ChannelHandshakeRelayer`.

The initial abstract implementation do not need to be as robust as in the v1 relayer. We will handle logic such as retry in separate issue. The initial implementation also do not need to pick up handshakes initiated by others, or resume from previous handshake.

In this issue, we will focus on a concrete implementation for Cosmos. The implementation of handshake for non-Cosmos chains will be handled in other issues.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
